### PR TITLE
HCPE-996 - Add pull request lifecycle docs

### PR DESCRIPTION
**Pull request by roaks3**
### :hammer_and_wrench: Description

Added docs to describe the PR lifecycle, meant primarily for new contributors.

### :building_construction: Acceptance tests

- [ ] Are there any feature flags that are required to use this functionality?
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: contributing/pull-request-lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Pull Request Submission and Lifecycle
```
Comment (roaks3) [inline contributing/pull-request-lifecycle.md:1]: Most of this content is sourced from other repos: [Kubernetes provider](https://raw.githubusercontent.com/hashicorp/terraform-provider-kubernetes/main/_about/PULL_REQUESTS.md), [AWS provider](https://raw.githubusercontent.com/hashicorp/terraform-provider-aws/main/docs/contributing/pullrequest-submission-and-lifecycle.md)

File: contributing/pull-request-lifecycle.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+__Note for internal service teams__: The pull request lifecycle below will be
+roughly the same for internal teams, but you will not need to fork the repo to
+create pull requests, and our recommendation is to get a review from one other
+member of your service team, in addition to a review from one member of the
+provider team.
```
Comment (roaks3) [inline contributing/pull-request-lifecycle.md:7]: This is the main note I added to address internal team members

File: contributing/pull-request-lifecycle.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+   either approve it or provide comments letting you know if there is anything
+   left to do. We do our best to keep up with the volume of PRs waiting for
+   review, but it may take some time depending on the complexity of the work.
+
+1. Once all outstanding comments and checklist items have been addressed, your
+   contribution will be merged! Merged PRs will be included in the next
+   provider release. The provider team takes care of updating the CHANGELOG and
+   releasing new versions of the provider. Please ask a provider team member
+   about the timing of the next release if the change you are making is
+   time-sensitive. Generally, releases are done roughly every two weeks.
```
Comment (roaks3) [inline contributing/pull-request-lifecycle.md:43]: These sentences at the end were added to set expectations for contributors on when their changes will be released (mainly aimed toward internal teams that have their own timelines to work with)

File: contributing/pull-request-lifecycle.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* For pull requests that follow the guidelines, we will proceed to reviewing
+  and merging, following the provider team's review schedule. There may be some
+  internal or community discussion needed before we can complete this.
+* Pull requests that don't follow the guidelines will be commented with what
+  they're missing. The person who submits the pull request or another community
+  member will need to address those requests before they move forward.
+
+## Pull Request Lifecycle
+
+1. [Fork the GitHub repository](https://help.github.com/en/articles/fork-a-repo),
```
Comment (bcmdarroch) [inline contributing/pull-request-lifecycle.md:21]: Might be worth linking to our resource checklist here.

**Review by bcmdarroch — APPROVED** [review]
This is great! 👍 👍

File: contributing/pull-request-lifecycle.md (hunk 4 above)
Comment (roaks3) [reply contributing/pull-request-lifecycle.md:21]: 👍 definitely agree. I forgot to mention this in the changes I had made, but I did add a link to our checklists in the next bullet.

**Pull request merged**